### ISSUE TITLE
New search editor should be pre-filled with selected text when pressing Ctrl + Shift + F

### ISSUE THREAD

**Issue by Merlin2001** [microsoft/vscode#116093]
I've been using the new `"search.mode": "newEditor"` setting as default lately and realized that it differs in an important way from `"search.mode": "view"`:  
In contrast to the search view, the search editor is **not initialized with the selected text** when pressing <kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>F</kbd>.

Even more inconsistently, when setting `"search.mode": "reuseEditor"` the re-used editor _is_ initialized with the selection (even ignoring the `seedSearchStringFromSelection` setting).

I think they should behave the same and potentially take into consideration #20768 so that <kbd>Ctrl</kbd> + <kbd>F</kbd> also behaves consistently.

I tested all combinations and they're a little inconsistent 🙈:

Pre-fill behavior legend:
`0`: Not pre-filled at all  
`1`: Pre-filled with selection only  
`2`: Pre-filled with selection or word the cursor is currently at (as criticized in #20768)

```
|  shortcut          |  seed..Selection |  search.mode  |  pre-fill behavior  |
|--------------------|------------------|---------------|---------------------|
|  Ctrl + F          |  false           |  irrelevant   |         0           |
|  Ctrl + F          |  true            |  irrelevant   |         2           |
|--------------------|------------------|---------------|---------------------|
|  Ctrl + Shift + F  |  false           |  view         |         0           |
|  Ctrl + Shift + F  |  false           |  reuseEditor  |         1 (!)       |
|  Ctrl + Shift + F  |  false           |  newEditor    |         0           |
|--------------------|------------------|---------------|---------------------|
|  Ctrl + Shift + F  |  true            |  view         |         1           |
|  Ctrl + Shift + F  |  true            |  reuseEditor  |         1           |
|  Ctrl + Shift + F  |  true            |  newEditor    |         0 (!)       |
```

**Comment by JacksonKearl** [comment]
Thanks for the detailed report! 

```
|  Ctrl + Shift + F  |  true            |  newEditor    |         0 (!)       |
```
This is a duplicate of https://github.com/microsoft/vscode/issues/115633 and is fixed in Insiders. 

```
|  Ctrl + Shift + F  |  false           |  reuseEditor  |         1 (!)       |
```
This is a bug. (and I suspect false/newEditor may behave incorrectly here Insiders too now)

**Comment by JacksonKearl** [comment]
Fixed 
```
|  Ctrl + Shift + F  |  false           |  reuseEditor  |         1 (!)       |
```

Left 
```
|  Ctrl + Shift + F  |  false           |  newEditor  |         1 (!)       |
```

as-is, as there's no damage to the new editor having a prefilled query.

**Comment by AndrewDavis** [comment]
So let me ask you this: I recently updated VSCode, and now I have this bug. As an end-user, what am I supposed to do about it? I tried installing Insiders edition, but I'd have to setup my environment all over again, so I uninstalled it.

This is a very obnoxious bug, and I don't wanna wait 1-2 months to get it fixed. So what do I do? Is there a rollback feature or something?

**Comment by ghiscoding** [comment]
wow seriously, I know this new bug is a little frustrating (I see it too and that is why I'm tracking this issue) but wow just calm down, this is an Open Source project, if you want this bug fix then pull up your sleeves and fix it and stop bashing at the VSCode team. You're not paying a single penny for it, so why are you writing such harsh message?? As an Open Source library maintainer, I see message like yours and it just makes me want to quit sometime... the least you could do is be polite and wait for a fix, especially since you didn't pay neither contribute to the project... or just move to a paid software where you can expect more.

**Comment by AndrewDavis** [comment]
> this is an Open Source project, if you want this bug fix then pull up your sleeves and fix it and stop bashing at the VSCode team.

I'm pretty sure they're getting paid. I have no familiarity with the codebase, and I'm not obligated to have it either, whether I'm satisfied with VSCode or not. I'm not bashing them, and if you say that I am, then one could just as easily say you're bashing me. But neither is the case, we're simply communicating.

> You're not paying a single penny for it, so why are you writing such harsh message??

It's not a harsh message.

> As an Open Source library maintainer, I see message like yours and it just makes me want to quit sometime... the least you could do is be polite and wait for a fix, especially since you didn't pay neither contribute to the project... or just move to a paid software where you can expect more.

Honestly if *that* makes you get emotional/upset and wanting to quit, you probably shouldn't be writing code. Code is cold, hard, scientific, and logical, and it impacts other peoples' ability to operate, for better or worse; if you write bad code, you're going to get negative feedback.

What you don't seem to understand here is this: My OS and the general tech world want me to keep my software up-to-date, so I updated it. As a result of this bug being introduced, the Find + Replace feature, something I use constantly for my *own* coding (which I myself do a lot of), was no longer functioning as it should, and was interrupting my creative process and progress. I attempted to remedy the situation, but I found no way to do so that would not be inhibitive and time-consuming. As a result, I gave them feedback about this, which is a necessary thing to do, or else nobody will know that people are having problems. Also, unless I ask these questions, which I had researched and could not find a good answer for, then I might never know what to do in this situation.

**Comment by AndrewDavis** [comment]
On a positive note, I found a temporary fix to this problem: change the `search.mode` setting ("Controls where new Search: Find in Files and Find in Folder operations occur: either in the sidebar's search view, or in a search editor") back to `view`. This way, selected text will be copied into the Search panel again.

**Comment by JacksonKearl** [comment]
@AndrewDavis I understand your compliant and I'm sorry you had trouble with this. To clarify: this feature was newly released last iteration (1.53.0) as an opt-in to different search modes. One of those modes (`newEditor`) did not behave the same as the existing default when text was selected in the editor. This was fixed in Insiders and will be fixed in Stable in 1-2 weeks. 

Based on your comments it appears that you opted-into this new behavior by setting the `search.mode` setting away from its default, the new behavior did not match your expectations, and opting back out by resetting the setting to its default value resolved this issue, is this correct?

I ask because it is not clear to me if you perhaps set this setting then forgot about it, or if this somehow got enabled by default -- the latter would be a severe bug I would need to look more in to, but I have not heard of any cases of that happening.

**Comment by AndrewDavis** [comment]
> Based on your comments it appears that you opted-into this new behavior by setting the `search.mode` setting away from its default, the new behavior did not match your expectations, and opting back out by resetting the setting to its default value resolved this issue, is this correct?

Correct, although I expected it would have been tested and the change would not be at risk for bugs like this, especially since it appears the same as the regular search panel. Thus, I didn't think anything of it, especially as being the cause/correlation of this bug.

Thanks for the fix.
